### PR TITLE
Refresh Data Sources (Part 1)

### PR DESCRIPTION
**Pull request by nlarew**
## Pull Request Info

[Outline](https://docs.google.com/document/d/1BWxOeeBx8GIr23C-KDQNoZqgPmjiVstyAmJPDIMqy0E/edit#)

- Ports & combines pages in MongoDB / Atlas Cluster & Data Lakes to match the outline structure
- Does not significantly modify the leftnav. This will happen in another PR.

### Jira

- (DOCSP-21891): CRUD & Aggregation APIs (Existing page)
- (DOCSP-21892): Read Preference (Existing page, rename)
- (DOCSP-21893): Wire Protocol (Combines 2 existing wire protocol pages + connection strings)
- (DOCSP-21894): Document Preimages (Existing page, rename)
- (DOCSP-21895): Internal Database Usage (Previously Overview > Considerations)

### Staged Changes (Requires MongoDB Corp SSO)

- [Read Preference](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fns-ds/mongodb/read-preference/)
- [Wire Protocol](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fns-ds/mongodb/wire-protocol/)
- [Document Preimages](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fns-ds/mongodb/preimages/)
- [Internal Database Usage](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fns-ds/mongodb/internal-database-usage/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/mongodb/internal-database-usage.txt [hunk 1]
```diff
@@ -0,0 +1,109 @@
+.. _internal-db-usage:
+
+=======================
+Internal Database Usage
+=======================
+
+{+service+} uses your linked MongoDB Atlas cluster to manage some
+internal operations. You may encounter these in some circumstances but
```
Comment (mongodben) [inline source/mongodb/internal-database-usage.txt:8]: "You may encounter these"...what do you mean by 'these' here? would be good if you could add a descriptive noun/phrase after 'these' to use the word as an adjective

File: source/mongodb/internal-database-usage.txt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+- The collection is now divergent: Sync clients do not see the new
+  document, but {+service+} clients do. 
+
+The same scenario holds true if a valid document is modified outside of
+{+sync+} such that one or more properties no longer conform to the Sync
+schema. 
+
+Any document that fails the object schema validation is added to the
+``__realm_sync.unsynced_documents`` collection (along with the reason
+for the failure) and will also be logged in the Realm UI ``Logs``
```
Comment (mongodben) [inline source/mongodb/internal-database-usage.txt:95]: would be good to show a schema of what this doc looks like, b/c based on this description seems that the unsynced document is only a sub-document of a larger document w/ some other metadata.

File: source/mongodb/preimages.txt [hunk 3]
```diff
@@ -15,33 +15,40 @@ View & Disable Collection-level Preimages
...
-.. tip::
+Enable Document Preimages
+-------------------------
 
-   For information on how to enable document preimages for a specific trigger,
-   see :ref:`Create a Database Trigger <create-a-database-trigger>`.
+Your app automatically enables document preimages for a collection when
+you define a trigger on that collection that uses preimages. Once you
+have enabled preimages for a collection, any trigger on the collection
+may enable document preimages with no additional overhead.
```
Comment (mongodben) [inline source/mongodb/preimages.txt:34]: this feels kinda redundant and i'm somewhat confused reading it. could you clarify?

File: source/mongodb/preimages.txt [hunk 4]
```diff
@@ -15,33 +15,40 @@ View & Disable Collection-level Preimages
...
+
+For information on how to enable document preimages for a specific
```
Comment (mongodben) [inline source/mongodb/preimages.txt:36]:
```suggestion
For information on how to enable document preimages for a
```

File: source/reference/service-limitations.txt [hunk 5]
```diff
@@ -143,9 +143,8 @@ Database Commands
 {+service-short+} does not support any :manual:`database commands
 </reference/command>` in the Client SDKs or :doc:`Functions
 </functions>`. You can, however, call a limited subset of database
-commands when when connected to a MongoDB cluster over the :doc:`{+service-short+}
-wire protocol </mongodb/connect-over-the-wire-protocol>`. For a list of
-wire protocol supported commands, see
+commands when when connected to a MongoDB cluster over the
+:ref:`wire-protocol`. For a list of supported commands, see
```
Comment (mongodben) [inline source/reference/service-limitations.txt:147]:
```suggestion
:ref:`wire protocol <wire-protocol>`. For a list of supported commands, see
```

nit personal preference. i find this shorthand to be kinda confusing/opaque.

File: source/sync/error-handling/errors.txt [hunk 6]
```diff
@@ -183,8 +183,7 @@ and MongoDB Atlas.
        Documents that do not match the schema cannot be synced and any local
        updates to the object represented by such a document will not propagate.
        
-       For more information, see 
-       :ref:`Realm Sync Schema Data Consistency <realm-sync-data-consistency>`.
+       For more information, see :ref:`unsynced-documents`.
```
Comment (mongodben) [inline source/sync/error-handling/errors.txt:186]: ditto above nit about named refs. as a fellow writer, i would like to know what text this would be showing.

File: source/sync/learn/overview.txt [hunk 7]
```diff
@@ -374,7 +374,7 @@ Realm Sync Schema Data Consistency
   If you create or modify {+sync+} documents using another tool, such as the 
   :binary:`~bin.mongo` shell or {+compass+}, be sure the documents validate 
   against the Sync object schema. For more information, see 
-  :ref:`Realm Sync Schema Data Consistency <realm-sync-data-consistency>`.
+  :ref:`unsynced-documents`.
```
Comment (mongodben) [inline source/sync/learn/overview.txt:377]: ditto above nit about named refs. as a fellow writer, i would like to know what text this would be showing.

**Comment by mongodben** [comment]
nice work. could you provide a brief overview of what content is ported where in the PR summary? it is quite hard to follow what's moved content, where it's moved to, and what is net-new content.

File: source/mongodb/preimages.txt [hunk 8]
```diff
@@ -15,33 +15,39 @@ View & Disable Collection-level Preimages
...
-.. tip::
+Enable Document Preimages
+-------------------------
 
-   For information on how to enable document preimages for a specific trigger,
-   see :ref:`Create a Database Trigger <create-a-database-trigger>`.
+Your app automatically enables document preimages for a collection when
+you define a trigger on that collection that uses preimages. Once you
+have enabled preimages for a collection, any trigger on the collection
+may use document preimages with no additional overhead.
```
Comment (mongodben) [inline source/mongodb/preimages.txt:34]:
```suggestion
may use document preimages.
```

not sure what you mean by this phrase. think it can be removed

**Review by mongodben — APPROVED** [review]
LGTM w 1 small suggestion. 

still unclear on how these changes fit within the larger epic, but trusting your judgement here.

**Comment by nlarew** [comment]
> still unclear on how these changes fit within the larger epic

Right now I'm moving to the new pages "in place" without large-scale toctree shuffling. In the last PR I'll update the sidebar to match the IA in the [outline](https://docs.google.com/document/d/1BWxOeeBx8GIr23C-KDQNoZqgPmjiVstyAmJPDIMqy0E/edit#) - the actual pages will already by done in previous commits.
